internal/history/service.go

```diff
@@ -48,19 +50,40 @@ type EvaluationHistoryService interface {
 	// in the history table.
 	ListEvaluationHistory(
 		ctx context.Context,
-		qtx db.Querier,
+		qtx db.ExtendQuerier,
 		cursor *ListEvaluationCursor,
 		size uint32,
 		filter ListEvaluationFilter,
 	) (*ListEvaluationHistoryResult, error)
 }
 
+type options func(*evaluationHistoryService)
+
+func withPropertiesServiceBuilder(builder func(qtx db.ExtendQuerier) propertiessvc.PropertiesService) options {
+	return func(ehs *evaluationHistoryService) {
+		ehs.propServiceBuilder = builder
+	}
+}
+
 // NewEvaluationHistoryService creates a new instance of EvaluationHistoryService
-func NewEvaluationHistoryService() EvaluationHistoryService {
-	return &evaluationHistoryService{}
+func NewEvaluationHistoryService(providerManager manager.ProviderManager, opts ...options) EvaluationHistoryService {
+	ehs := &evaluationHistoryService{
+		providerManager: providerManager,
+		propServiceBuilder: func(qtx db.ExtendQuerier) propertiessvc.PropertiesService {
+			return propertiessvc.NewPropertiesService(qtx)
```

Comment:
I think it would be OK to just require the PropertiesService interface as a parameter but the options don't hurt..

Reply:
It was done for testing only.